src/Text/Parsing/StringParser/CodePoints.purs

```diff
@@ -46,17 +50,23 @@ eof = Parser \s ->
     { substring, position } | 0 < SCP.length substring -> Left { pos: position, error: "Expected EOF" }
     _ -> Right { result: unit, suffix: s }
 
--- | Match any character.
+-- | Match any character from the Basic Multilingual Plane.
 anyChar :: Parser Char
-anyChar = Parser \{ substring, position } ->
-  case SCP.codePointAt 0 substring of
-    Just cp -> case toChar cp of
-      Just chr -> Right { result: chr, suffix: { substring: SCP.drop 1 substring, position: position + 1 } }
-      Nothing -> Left { pos: position, error: "CodePoint " <> show cp <> " is not a character" }
-    Nothing -> Left { pos: position, error: "Unexpected EOF" }
+anyChar = do
+  cp <- anyCodePoint
+  case toChar cp of
+    Just chr -> pure chr
+    Nothing -> fail $ "Code point " <> show cp <> " is not a character"
   where
   toChar = fromCharCode <<< fromEnum
 
+-- | Match any code point.
+anyCodePoint :: Parser CodePoint
+anyCodePoint = Parser \{ substring, position } ->
+  case SCP.codePointAt 0 substring of
```

Comment:
This looks fine but you could use `uncons` like so https://github.com/purescript-contrib/purescript-parsing/blob/33cbb663651f28062a2ffc0016033142b24bfed2/src/Text/Parsing/Parser/String.purs#L102

Reply:
That's better indeed